packages/@okta/vuepress-site/code/java/spring_security_saml/index.md

```diff
@@ -7,349 +7,375 @@ excerpt: >-
-
-@Configuration
-public class SecurityConfiguration {
-
-    @Bean
-    SecurityFilterChain configure(HttpSecurity http) throws Exception {
-
-        OpenSaml4AuthenticationProvider authenticationProvider = new OpenSaml4AuthenticationProvider();
-        authenticationProvider.setResponseAuthenticationConverter(groupsConverter());
-
-        // @formatter:off
-        http
-            .authorizeHttpRequests(authorize -> authorize
-                .mvcMatchers("/favicon.ico").permitAll()
-                .anyRequest().authenticated()
-            )
-            .saml2Login(saml2 -> saml2
-                .authenticationManager(new ProviderManager(authenticationProvider))
-            )
-            .saml2Logout(withDefaults());
-        // @formatter:on
-
-        return http.build();
-    }
-
-    private Converter<OpenSaml4AuthenticationProvider.ResponseToken, Saml2Authentication> groupsConverter() {
-
-        Converter<ResponseToken, Saml2Authentication> delegate =
-            OpenSaml4AuthenticationProvider.createDefaultResponseAuthenticationConverter();
-
-        return (responseToken) -> {
-            Saml2Authentication authentication = delegate.convert(responseToken);
-            Saml2AuthenticatedPrincipal principal = (Saml2AuthenticatedPrincipal) authentication.getPrincipal();
-            List<String> groups = principal.getAttribute("groups");
-            Set<GrantedAuthority> authorities = new HashSet<>();
-            if (groups != null) {
-                groups.stream().map(SimpleGrantedAuthority::new).forEach(authorities::add);
-            } else {
-                authorities.addAll(authentication.getAuthorities());
-            }
-            return new Saml2Authentication(principal, authentication.getSaml2Response(), authorities);
-        };
-    }
-}
-```
-
-Finally, modify your `build.gradle` file to force the latest version of Open SAML that works with Spring Security 6.
-
-```groovy
-repositories {
-    ...
-    maven { url "https://build.shibboleth.net/nexus/content/repositories/releases/" }
-}
-
-dependencies {
-    constraints {
-        implementation "org.opensaml:opensaml-core:4.1.1"
-        implementation "org.opensaml:opensaml-saml-api:4.1.1"
-        implementation "org.opensaml:opensaml-saml-impl:4.1.1"
-    }
-    ...
-}
-```
-
-Now, if you restart your app and log in, you should see your user's groups as authorities. _Huzzah!_
-
-![Groups as authorities](/img/spring-security-saml-groups-as-authorities.png "Groups as authorities")
-
-## Next Steps
-
-At this point, you should be familiar with setting up SAML-enabled applications to work with an Okta organization and how to configure Spring Security SAML to work with Okta.
+When you sign in, the resulting page shows that you have a `ROLE_USER` authority. However, when you assign users to the app, you give access to `Everyone`. You can configure your SAML app on Okta to send a user's groups as an attribute, and add other attributes like name and email.
```

Comment:
It seems like this sentence:

> However, when you assign users to the app, you give access to `Everyone`

Should be:

> However, when you assigned users to the app, you gave access to `Everyone`

Reply:
Updated in next commit